core/local/local.go

```diff
@@ -57,33 +57,26 @@ func (h *vuHandle) run(logger *logrus.Logger, flow <-chan int64, iterDone chan<-
 	h.RUnlock()
 
 	for {
-		select {
-		case _, ok := <-flow:
-			if !ok {
-				return
-			}
-		case <-ctx.Done():
+		if _, ok := <-flow; !ok || ctx.Err() != nil {
```

Comment:
This seems wrong - you're changing the logic, the new code doesn't do what the old code did.

Reply:
Just curiously where is the difference?

Reply:
I hope this answers your question :wink: https://play.golang.org/p/N6KzvGivmgB